### DISCUSSION TITLE
linux下demo无法编译成功

### DISCUSSION THREAD

**Discussion by rebornfz** [Tencent/libpag#271] [Q&A] [answered]
在Linux上拉取了main的主干最新分支，可以顺利编译，但demo无法make成功
`[root@0fbc7398be40 build]# cmake ..
--begin complie---
-- Configuring done
-- Generating done
-- Build files have been written to: /home/libpag/linux/build
[root@0fbc7398be40 build]# make
[ 50%] Linking CXX executable pag-linux
CMakeFiles/pag-linux.dir/src/main.cpp.o: In function `ReplaceImageOrText()':
main.cpp:(.text+0xfb): undefined reference to `pag::PAGFile::Load(std::string const&)'
main.cpp:(.text+0x1ad): undefined reference to `pag::PAGImage::FromPath(std::string const&)'
main.cpp:(.text+0x31e): undefined reference to `pag::PAGFont::RegisterFont(std::string const&, int, std::string const&, std::string const&)'
collect2: 错误：ld 返回 1
make[2]: *** [pag-linux] 错误 1
make[1]: *** [CMakeFiles/pag-linux.dir/all] 错误 2
make: *** [all] 错误 2`

**Comment by kevingpqi123** [comment, marked as answer]
更新下最新代码，CMake配置问题

**Reply by rebornfz** [reply]
可以编译了，但是运行出错，读pag出错，---pagFile is nullptr!!!
判断了pag文件是存在的

**Reply by rebornfz** [reply]
@kevingpqi123 
有测试了一下，byte_data =pag::ByteData::FromPath可以成功，然后pag::PAGFile::Load(byte_data->data(), byte_data->length())失败了

**Reply by kevingpqi123** [reply]
建议你先直接运行linux的demo工程，先保证demo是可以正常运行的，如果运行过程中报---pagFile is nullptr!!!，大概率是你没有执行git lfs pull，pag文件没有被完整拉取下来
